libs/product/driver/magento/src/product.service.spec.ts

```diff
@@ -135,7 +135,7 @@ describe('Product | Magento | ProductService', () => {
 
       const op = controller.expectOne(addTypenameToDocument(getProductByUrl()));
 
-      expect(op.operation.variables.url).toEqual('TESTING_URL');
+      expect(op.operation.variables.url).toEqual('path/to/TESTING_URL');
```

Comment:
@damienwebdev we shouldn't need to truncate leading path segments for products, right? I can't remember why this behavior was necessary

Reply:
bump^

Reply:
I believe this does matter since Magento's GraphQl product lookup is by url_key which does not include category paths for example.

Think:

`furniture/chairs/brown-chair.html`

vs.

`brown-chair.html`.

This, at least is how the current driver works. Perhaps we could use the `route` node instead?

Reply:
So `DaffMagentoProductService.getByUrl` should successfully return for `furniture/chairs/brown-chair.html` even though the platform doesn't recognize that as a product?

Reply:
Well, it is the `url` of a specific product, albeit with some leading paths.

Reply:
changed to use the route query